minimal.js

```diff
@@ -83,36 +82,50 @@ module.exports = function (dirname, keys, opts) {
 
   var append = db.rawAppend = db.append
   db.post = Obv()
-  var queue = AsyncWrite(function (_, cb) {
-    var batch = state.queue
-    state.queue = []
-    append(batch, function (err, v) {
+
+  let writing = false
+
+  const write = () => {
```

Comment:
:fire: 

I would rather have `if (writing) return` at the top of this to defend against accidental multiple calls. 

You can then replace `if (writing === false) write()` with just `write()` below

Reply:
`<style>` I think I'd also be tempted to staple the isWriting variable to this function like `write.isRunning`

Reply:
> I would rather have `if (writing) return` at the top of this to defend against accidental multiple calls.

I thought so too, but then `write()` can't call itself without setting `writing = false`. If you do that then both `queue()` and `write()` both call `write()`, and although only one of them succeeds you still get duplicate function calls. Maybe you could propose a patch that you think would do this better?

> I think I'd also be tempted to staple the isWriting variable to this function like `write.isRunning`

I'd be comfortable seeing this implemented as a 'writer' class that holds the SSB-Validate state and writing state internally, but I have a much harder time understanding code that staples properties onto random variables that aren't objects.

Reply:
this makes sense, happy to leave as is.
Would be nice to collect state together as much as possible, perhaps that's a different patch though